### ISSUE TITLE
A newly emitting loss function is loose on both parameters, for two different reasons

### ISSUE THREAD

**Issue by khatchad** [ponder-lab/Hybridize-Functions-Refactoring#912]
Tracking issue for a newly emitting function whose specification is loose on both parameters, and for the two distinct reasons it is loose. Filed so the looseness has a home: the function began emitting only with the Ariadne 0.52.85 bump (#911), so nothing recorded it before.

**The emitted shapes are deliberately not restated here.** They are being adjudicated independently and from the source alone, and reading our output first would anchor that judgment. They will be added once that lands.

## The Function

```python
model = tf.keras.Sequential([featureExtractor, tf.keras.layers.Dense(NUM_CLASS, activation="softmax")])

train_generator = train_datagen.flow_from_directory(
        os.path.join(project_directory, "./mnist/train"),
        target_size=(IMG_SIZE, IMG_SIZE),
        batch_size=BATCH_SIZE,
        class_mode="categorical")


def compute_loss(labels, predictions):
    per_example_loss = loss_object(labels, predictions)
    return tf.nn.compute_average_loss(per_example_loss, global_batch_size=BATCH_SIZE)
```

TensorFlow2.0-Examples, `7-Utils/multi_gpu_train.py:98`. Both parameters carry the same extent at run time: the classification width. They reach it by different routes, and only one of those routes is one this tool could follow.

## Two Bounds, Two Owners

**`predictions` is bounded by an unmodeled forward shape.** It is the result of calling a `Sequential`-built model, whose final layer is `Dense(NUM_CLASS)`. The width is therefore present in the source, and recovering it is analysis work: wala/ML#832 records that a `Sequential`-built model resolves no forward shape through its layer list. Nothing is owed here beyond that issue; when it lands, this parameter tightens on its own.

**`labels` is bounded by evidence that does not exist statically.** It arrives from the generator, which is given a directory, a target size, a batch size and `class_mode="categorical"`. It is not given a `classes` argument, and the module's `NUM_CLASS` constant reaches only the `Dense` layer, never the generator. The width is the data directory's subdirectory count at run time, and that directory is not part of the project. So an unresolved axis is the correct answer on the evidence available, not a precision failure, and no amount of analysis work closes it. Closing it is an annotation-level act.

## Why This Is Filed Rather Than Left To The Adjudication

A record of what the tight specification *is* does not track either the work of closing a gap or the decision to accept one. The second parameter's gap in particular is not closable by analysis at all, so it can only ever be closed by writing an annotation or accepted deliberately. An acceptance that is never written down is indistinguishable from an oversight, which is the whole reason to have somewhere to write it.

## Disposition To Settle

1. Whether the second parameter's axis is accepted, with the reason recorded, or an annotation is written for it. Note the extent itself is unverifiable from the project as committed, since the data directory is absent.
2. Nothing for the first parameter beyond tracking wala/ML#832.

Blocked on neither; both are decisions rather than work.

**Comment by khatchad** [comment]
## The Measured Emission

Withheld from the original report on purpose, because the tightest specification for this function was being determined independently from the source and reading our output first would have anchored that judgment. That work is done, so here is what we emit, on Ariadne 0.52.85:

| Parameter | Emitted | Bound |
| --- | --- | --- |
| `labels` | `TensorSpec(shape=(None, None), dtype=tf.float32)` | The classification width has no static evidence channel |
| `predictions` | `TensorSpec(shape=None, dtype=tf.float32)` | Unknown rank: wala/ML#832 |

The second is not a loose shape but the absence of one. A `shape=None` entry is unknown rank, the weakest specification that still runs, and it is what a `Sequential`-built model's call yields today: a definite dtype and no shape at all.

Both are runnable. Executed on the pinned TensorFlow, the emitted pair runs and returns what the undecorated function returns, so nothing here is a correctness defect. The gap is precision, and the two halves have the different owners this issue records.

## What Changes Each

On `predictions`, nothing is owed on this side. When wala/ML#832 lands, the forward shape resolves through the layer list, `Dense(NUM_CLASS)` supplies the width, and the emission tightens with no consumer work.

On `labels`, nothing closes it by analysis, ever. The width is a run-time property of a directory the project does not contain. Either an annotation supplies it, and asserts something unverifiable from the repository as committed, or the axis is left unresolved deliberately and that decision is recorded here.
